lib/rules/arrow-body-style.js

```diff
@@ -65,38 +68,57 @@ module.exports = {
             const arrowBody = node.body;
 
             if (arrowBody.type === "BlockStatement") {
-                if (never) {
+                const blockBody = arrowBody.body;
+
+                if (blockBody.length !== 1 && !never) {
+                    return;
+                }
+
+                if (asNeeded && requireReturnForObjectLiteral && blockBody[0].type === "ReturnStatement" &&
+                    blockBody[0].argument.type === "ObjectExpression") {
```

Comment:
`ReturnStatement#argument` property needs null check.
Chould you add tests like `a => { return }`?